src/components/upload/Home.tsx

```diff
@@ -31,9 +34,14 @@ const createTree = async ({
   return btoa(result.key);
 };
 
+const hasFinished = (progress: { [hash: string]: number }): boolean => {
+  const total = Object.values(progress).reduce((a, b) => a + b, 0);
+  return total === Object.keys(progress).length * 100;
+};
+
 const Home: FC<{}> = () => {
-  // TODO: @jdalzatec Bring progress here
-  const { files } = useContext(FileContext);
+  const { files, progress } = useContext(FileContext);
+  const finished = hasFinished(progress);
```

Comment:
I think we can continue if all these `files` have finished, no need to check all the progress array that might contain deleted files.

Reply:
This was taken into account. Thanks!